File: spotifyarch/src/main/java/com/timothy/spotifyarchitecture/room/ServiceComponent.java
package com.timothy.spotifyarchitecture.room;

import javax.inject.Singleton;

import dagger.Component;

/**
 * Created by tim on 5/31/17.
 */
@Singleton
@Component(modules = SpotifyModule.class)
public interface ServiceComponent {
    //void inject(MainActivity mainActivity);
}

File: app/src/main/kotlin/com/timmahh/musicall/ServiceComponent.kt
package com.timmahh.musicall

import com.timmahh.spotifyweb.presenters.SpotifyDrawerLayout
import dagger.Component
import javax.inject.Singleton

/**
 * Creates a Dagger 2.0 component interface to inject MainActivity
 */
@Singleton
@Component(modules = arrayOf(ServiceModule::class))
interface ServiceComponent {
    fun inject(mainActivity: MainActivity)
    fun inject(spotifyDrawerLayout: SpotifyDrawerLayout)
}
